src/util_libc.rs

```diff
@@ -20,6 +20,12 @@ cfg_if! {
         use libc::__error as errno_location;
     } else if #[cfg(target_os = "haiku")] {
         use libc::_errnop as errno_location;
+    } else if #[cfg(target_os = "horizon")] {
```

Comment:
I think you should use the `cfg` as in lib.rs, i.e. add `target_arch = "arm"`.

Reply:
If you wanted it to be a bit more generic, I think it could even be `target_env = "newlib"`, although perhaps more specific is better in the interest of avoiding breakage for other targets.

Reply:
Updated